psi4/src/psi4/libmints/osrecur.cc

```diff
@@ -1234,6 +1234,1129 @@ void ObaraSaikaTwoCenterEFPRecursion::compute(double PA[3], double PB[3], double
     delete[] F;
 }
 
+ObaraSaikaTwoCenterMultipolePotentialRecursion::ObaraSaikaTwoCenterMultipolePotentialRecursion(int max_am1, int max_am2,
+                                                                                               int max_k)
+    : max_am1_(max_am1), max_am2_(max_am2) {
+    if (max_am1 < 0)
+        throw SanityCheckError("ERROR: ObaraSaikaTwoCenterMVIRecursion -- max_am1 must be nonnegative", __FILE__,
+                               __LINE__);
+    if (max_am2 < 0)
+        throw SanityCheckError("ERROR: ObaraSaikaTwoCenterMVIRecursion -- max_am2 must be nonnegative", __FILE__,
+                               __LINE__);
+
+    do_dipole = (max_k >= 1);
+    do_quadrupole = (max_k >= 2);
+    do_octupole = (max_k >= 3);
+    size_ = max_am1 > max_am2 ? max_am1 : max_am2;
+    size_ += 1;
+    size_ = (size_ - 1) * size_ * (size_ + 1) + 1;
+    q_ = init_box(size_, size_, max_am1_ + max_am2_ + 4);
+    x_ = init_box(size_, size_, max_am1_ + max_am2_ + 3);
+    y_ = init_box(size_, size_, max_am1_ + max_am2_ + 3);
+    z_ = init_box(size_, size_, max_am1_ + max_am2_ + 3);
+    xx_ = init_box(size_, size_, max_am1_ + max_am2_ + 2);
```

Comment:
Is it worth it to only init the ones required?

Reply:
Done.